pkg/accountdb/repositories/user.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package repositories
+
+import (
+	"context"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"storj.io/storj/pkg/accountdb/dbo"
+	"storj.io/storj/pkg/accountdb/dbx"
+)
+
+// Exposes methods to manage User table in database.
+type User interface {
+	GetByCredentials(password, email string) (*dbo.User, error)
+	Get(id uuid.UUID) (*dbo.User, error)
+	Insert(user *dbo.User) (error)
+	Delete(id uuid.UUID) (error)
+	Update(user *dbo.User) (error)
+}
+
+// implementation of User interface repository using spacemonkeygo/dbx orm
+type user struct {
+	db *dbx.DB
+	ctx context.Context
+}
+
+// Constructor for user repository
+func NewUserRepository(db *dbx.DB, ctx context.Context) User {
+	return &user{
+		db,
+		ctx,
+	}
+}
+
+// Method for querying user by id from the database.
+func (u *user) Get(id uuid.UUID) (*dbo.User, error) {
+
+	result := &dbo.User{}
+
+	userId := dbx.User_Id(id.String())
+
+	user, err :=  u.db.Get_User_By_Id(u.ctx, userId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result.FromDbx(user)
+}
+
+// Method for querying user by credentials from the database.
+func (u *user) GetByCredentials(password, email string) (*dbo.User, error) {
+
+	result := &dbo.User{}
+
+	userEmail := dbx.User_Email(email)
+	userPassword := dbx.User_PasswordHash(password)
```

Comment:
actually hash the password here

Reply:
not sure that it should be done in data access layer